### PR TITLE
Move the templates and profiles api implementation into on-taskgraph

### PR DESCRIPTION
**Pull request by tannoa2**
Move template and profile APIs to on-taskgraph

Signed-off-by: Kranti Uppala <kranti.uppala@dell.com>

jenkins: depends on https://github.com/RackHD/on-http/pull/787, https://github.com/RackHD/on-core/pull/308, https://github.com/RackHD/RackHD/pull/894

### CODE REVIEW

**Comment by tannoa2** [comment]
jenkins: ignore

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

File: spec/lib/services/profile-api-service-spec.js [hunk 1]
```diff
@@ -486,7 +474,7 @@ describe("Http.Services.Api.Profiles", function () {
 
             return profileApiService.getProfiles(req, query, res)
                 .then(function () {
-                    expect(waterline.nodes.findByIdentifier).to.have.been.calledOnce;
+                    // expect(waterline.nodes.findByIdentifier).to.have.been.calledOnce;
```
Comment (geoff-reid) [inline spec/lib/services/profile-api-service-spec.js]: remove this line

**Pull request merged**